index.d.ts

```diff
@@ -0,0 +1,21 @@
+interface UnusedFilename {
+	/**
+	 * Get an unused filename by appending a number if it exists: `file.txt` → `file (1).txt`.
+	 *
+	 * @param filepath - The path to check for filename collision.
+	 * @returns Either the original `filename` or the `filename` appended with a number.
+	 */
+	(filepath: string): Promise<string>;
+
+	/**
+	 * Synchronpusly get an unused filename by appending a number if it exists: `file.txt` → `file (1).txt`.
+	 *
+	 * @param filepath - The path to check for filename collision.
+	 * @returns Either the original `filename` or the `filename` appended with a number.
+	 */
+	sync(filepath: string): string;
+}
+
+declare const unusedFilename: UnusedFilename;
```

Comment:
I prefer the style of not using a named interface when not needed: https://github.com/sindresorhus/image-type/blob/333daed89e27a7ca967ff88221ce959ca4e78cd6/index.d.ts#L40